src/test/java/itest/TargetPostDeleteIT.java

```diff
@@ -0,0 +1,229 @@
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.add("connectUrl", "invalidConnectUrl");
+
+        webClient
+                .post(REQ_URL)
+                .sendForm(
+                        form,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                        });
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(
+                ((HttpStatusException) ex.getCause()).getStatusCode(), Matchers.equalTo(400));
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
+    }
+
+    @Test
+    public void testPostTargetThrowsWithDuplicateConnectUrl() throws Exception {
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.add("alias", "duplicateCryostat");
+        form.add("connectUrl", SELF_REFERENCE_TARGET_ID);
+
+        webClient
+                .post(REQ_URL)
+                .sendForm(
+                        form,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                        });
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(
+                ((HttpStatusException) ex.getCause()).getStatusCode(), Matchers.equalTo(400));
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
+    }
+
+    @Test
+    public void testDeleteTargetThrowsWithInvalidConnectUrl() throws Exception {
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.add("targetId", "invalidConnectUrl");
+
+        webClient
+                .post(REQ_URL)
+                .sendForm(
+                        form,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                        });
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(
+                ((HttpStatusException) ex.getCause()).getStatusCode(), Matchers.equalTo(400));
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
+    }
+
+    @Test
+    public void testDeleteTargetThrowsWithNonExistentConnectUrl() throws Exception {
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        final String nonExistentUrl =
+                URLEncodedUtils.formatSegments(
+                        String.format("service:jmx:rmi:///jndi/rmi://invalid:9091/jmxrmi"));
```

Comment:
I keep getting a `400` when trying different variations of JMX service URLs or `hostname:port`. How can I trigger a `404`?

Reply:
Relevant code in `TargetDeleteHandler`:
```
String targetId = params.getPathParams().get("targetId");
            if (StringUtils.isBlank(targetId)) {
                throw new ApiException(400, "Invalid targetId");
            }
            URI uri = URIUtil.createAbsolute(targetId);
            if (!customTargetPlatformClient.removeTarget(uri)) {
                throw new ApiException(404);
            }
```

Reply:
I think the issue is that you're POSTing a form with the targetId - it should be a DELETE verb, and the request path should include the targetId as a path parameter.

Reply:
Whoops, thanks